cmds/hwclock/hwclock.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// hwclock sets or reads the hwclock in UTC format.
+//
+// Synopsis:
+//     hwclock [-w]
+//
+// Description:
+//     It prints the current hwclock time in UTC if called without any flags.
+//     It sets the hwclock to the system clock in UTC if called with -w.
+//
+// Options:
+//     -w: set hwclock to system clock in UTC
+//
+// Author:
+//     claudiozumbo@gmail.com.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"log"
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+const startYear = 1900
+
+var write = flag.Bool("w", false, "Set hwclock from system clock in UTC")
+
+func readRtc(rtcFile *os.File) (time.Time, error) {
+
+	var (
+		ptm     unix.RTCTime
+		ret time.Time
+	)
+
+	_, _, errno := syscall.Syscall(syscall.SYS_IOCTL,
+		uintptr(rtcFile.Fd()),
+		uintptr(unix.RTC_RD_TIME),
```

Comment:
`golang.org/x/sys/unix` now also has a wrapper for this: https://godoc.org/golang.org/x/sys/unix#IoctlGetRTCTime. It was added by golang/sys@aca44879d5644da7c5b8ec6a1115e9b6ea6c40d9, so the vendored copy would need to be updated.